.evergreen/config.yml

```diff
@@ -21,8 +21,23 @@ functions:
       params:
         script: |-
           set -o errexit
-          chmod u+x libmongocrypt/.evergreen/*.sh
-          ./libmongocrypt/.evergreen/print-env-info.sh
+          cd libmongocrypt
+          chmod u+x .evergreen/*.sh
+          ./.evergreen/print-env-info.sh
+
+          if [ "${is_patch}" != "true" ]; then
+            # determine if we have a release tag present on HEAD
+            head_tag=$(git tag -l --points-at HEAD '[0-9].*' | tail -1)
```

Comment:
Given `HEAD` points to a commit with the following tags:
- `1.2.3`
- `1.2.3-pre`
- `some-tag`
The command `git tag --points-at HEAD '[0-9].*' | tail -1` yields the prerelease version `1.2.3-pre` on my local machine. I _think_ the intent of `tail -1` is to select `1.2.3` instead? If so, this command may need to be updated with a `sort` accordingly for deterministic behavior.

Reply:
The issue with `--sort` is that it is not present on some older version of Git.  In principle, you are correct.  However, in practice the release workflow requires a separate commit for every release.  That is to say, we will never release by going to the commit already tagged `x.y.z-rcW` and then re-tagging it `x.y.z` and pushing only the tag.

For a related example, have a look at the history of `build/calc_release_version.py` in the C driver and also at CDRIVER-2831.  In that instance, because we were working in Python it was easily possible to implement the sorting behavior in the application code.  Here, since we are working in shell, we'd have to resort to `sort` (or perhaps `sort -r`).

After typing all that I see that you might have been referring to the `sort` command all along and not to the `--sort` option to Git.  It doesn't seem strictly necessary here, but it certainly won't hurt and it would make things more robust.